ambry-tools/src/main/java/com.github.ambry/store/DumpDataHelper.java

```diff
@@ -260,6 +265,9 @@ LogBlobRecordInfo readSingleRecordFromLog(RandomAccessFile randomAccessFile, lon
     } else {
       throw new MessageFormatException("Header version not supported " + version, MessageFormatErrorCodes.IO_Error);
     }
+    if(throttler != null){
+      throttler.maybeThrottle(totalRecordSize);
```

Comment:
It is a bit confusing, since the maybeThrottle api has parameter `observed` that is "the newly observed units since the last time this method was called", and here `totalRecordSize` is given.